deepfence_worker/cronjobs/console.go

```diff
@@ -2,56 +2,40 @@ package cronjobs
 
 import (
 	"context"
-	"os"
-	"strconv"
 
 	"github.com/deepfence/ThreatMapper/deepfence_server/controls"
 	utils_ctl "github.com/deepfence/ThreatMapper/deepfence_utils/controls"
+	"github.com/deepfence/ThreatMapper/deepfence_utils/directory"
 	"github.com/deepfence/ThreatMapper/deepfence_utils/log"
 	ctl "github.com/deepfence/ThreatMapper/deepfence_worker/controls"
 	"github.com/hibiken/asynq"
 )
 
 const (
-	ConsoleAgentId     = "deepfence-console-cron"
-	DefaultMaxWorkload = 5
+	ConsoleAgentId      = "deepfence-console-cron"
+	ContextAllocatorKey = "scan-workload-allocator"
 )
 
-var (
-	MaxWorkload           int
-	ScanWorkloadAllocator *utils_ctl.WorkloadAllocator
-)
-
-func init() {
-	numWorkloadStr := os.Getenv("DEEPFENCE_MAX_SCAN_WORKLOAD")
-	if len(numWorkloadStr) == 0 {
-		MaxWorkload = DefaultMaxWorkload
-	} else {
-		numWorkload, err := strconv.Atoi(numWorkloadStr)
-		if err != nil {
-			MaxWorkload = DefaultMaxWorkload
-		} else {
-			MaxWorkload = numWorkload
-		}
-	}
-	ScanWorkloadAllocator = utils_ctl.NewWorkloadAllocator(DefaultMaxWorkload)
-}
-
 /*
-While this functon is a cron job, it is running on the worker's address space
-Hence Allocator can be shared across tasks
+Allocator shared across all workers instances per namespace using redis counter
 */
 func TriggerConsoleControls(ctx context.Context, t *asynq.Task) error {
-	log.Debug().Msgf("Trigger console actions #capacity: %v", ScanWorkloadAllocator.MaxAllocable())
+	ScanWorkloadAllocator := ctx.Value(ContextAllocatorKey).(*utils_ctl.RedisWorkloadAllocator)
+
+	ns, _ := directory.ExtractNamespace(ctx)
+
+	allocatable := ScanWorkloadAllocator.MaxAllocable()
 
-	actions, errs := controls.GetAgentActions(ctx, ConsoleAgentId, int(ScanWorkloadAllocator.MaxAllocable()))
+	actions, errs := controls.GetAgentActions(ctx, ConsoleAgentId, int(allocatable))
 	for _, e := range errs {
 		if e != nil {
 			log.Error().Msgf(e.Error())
 		}
 	}
+	log.Info().Str("namespace", string(ns)).
+		Msgf("Trigger console actions #capacity: %v got #actions: %d", allocatable, len(actions))
```

Comment:
Move this before we do `GetAgentActions` 
Otherwise it will appear after errors